src/Lexer.js

```diff
@@ -230,9 +242,30 @@ module.exports = class Lexer {
       }
 
       // top-level paragraph
-      if (top && (token = this.tokenizer.paragraph(src))) {
+      // prevent paragraph consuming extensions by clipping 'src' to extension start
+      cutSrc = src;
+      if (this.options.extensions?.startBlock) {
+        let startIndex = Infinity;
+        const tempSrc = src.slice(1);
+        let tempStart;
+        this.options.extensions.startBlock.forEach(function(getStartIndex) {
+          tempStart = getStartIndex(tempSrc);
```

Comment:
furthermore I know that Array.forEach is a lot slower than a for loop. I don't think that matters a whole lot since we are not guaranteeing speed with extensions but it is something to think about.

Reply:
> Array.forEach is a lot slower than a for loop

Ok. I'm immediately worried since I don't expect this array to ever get that huge but I might take a look through all the loops I added and switch them out later.

Reply:
Done.

Reply:
> I don't expect this array to ever get that huge

true